### PR TITLE
BXC-2527 - MODS History/versioning

### PR DESCRIPTION
**Pull request by bbpennel**
https://jira.lib.unc.edu/browse/BXC-2527

* Implements service for updating files which are versioned
* Implements datastream history log for adding textual contents to a history log
* Adds namespace for packaging xml
* Removes contentObject.setDescription in favor of UpdateDescriptionService
  * Service now adds description relations, transfers binaries, employs versioning, and is more configurable
* IngestContentObjectJob now uses UpdateDescriptionService for setting mods
  * Transfer of mods now takes place during this job
* Adds method for getting creation timestamp of objects

### CODE REVIEW

File: persistence/src/test/java/edu/unc/lib/dl/persist/services/edit/UpdateDescriptionServiceTest.java [hunk 1]
```diff
@@ -114,34 +117,49 @@ public void init() throws Exception{
...
-        verify(transferSession).transferReplaceExisting(eq(modsPid), inputStreamCaptor.capture());
-        assertEquals(FILE_CONTENT, IOUtils.toString(inputStreamCaptor.getValue()));
+        verify(versioningService).addVersion(versionCaptor.capture());
+        DatastreamVersion version = versionCaptor.getValue();
+        assertEquals(modsPid, version.getDsPid());
+        assertEquals("text/xml", version.getContentType());
+        assertEquals(FILE_CONTENT, IOUtils.toString(version.getContentStream()));
 
-        verify(obj).setDescription(transferredUri);
+        verify(versioningService).addVersion(any(DatastreamVersion.class));
```
Comment (bbpennel) [inline persistence/src/test/java/edu/unc/lib/dl/persist/services/edit/UpdateDescriptionServiceTest.java]: Is this needed?

File: persistence/src/test/java/edu/unc/lib/dl/persist/services/edit/UpdateDescriptionServiceTest.java [hunk 2]
```diff
@@ -114,34 +117,49 @@ public void init() throws Exception{
...
+
+        service.updateDescription(agent, objPid, modsStream);
 
+        verify(versioningService).addVersion(versionCaptor.capture());
+        DatastreamVersion version = versionCaptor.getValue();
+        assertEquals(modsPid, version.getDsPid());
+        assertEquals("text/xml", version.getContentType());
+        assertEquals(FILE_CONTENT, IOUtils.toString(version.getContentStream()));
+
+        verify(versioningService).addVersion(any(DatastreamVersion.class));
```
Comment (bbpennel) [inline persistence/src/test/java/edu/unc/lib/dl/persist/services/edit/UpdateDescriptionServiceTest.java]: check

File: persistence/src/test/java/edu/unc/lib/dl/persist/services/edit/UpdateDescriptionServiceTest.java [hunk 3]
```diff
@@ -171,14 +202,31 @@ public void invalidModsValidationOff() throws Exception {
...
+
+        service.updateDescription(agent, objPid, modsStream);
+
+        verify(versioningService).addVersion(versionCaptor.capture());
+        DatastreamVersion version = versionCaptor.getValue();
+        assertEquals(modsPid, version.getDsPid());
+        assertEquals("text/xml", version.getContentType());
+        assertEquals(FILE_CONTENT, IOUtils.toString(version.getContentStream()));
+
+        verify(versioningService).addVersion(any(DatastreamVersion.class));
```
Comment (bbpennel) [inline persistence/src/test/java/edu/unc/lib/dl/persist/services/edit/UpdateDescriptionServiceTest.java]: check

File: services/src/test/resources/retrieve-mods-it-servlet.xml [hunk 4]
```diff
@@ -32,6 +32,25 @@
 
     <context:component-scan resource-pattern="**/RetrieveMODSController*" base-package="edu.unc.lib.dl.cdr.services.rest"/>
     
+    <bean id="versionedDatastreamService" class="edu.unc.lib.dl.persist.services.versioning.VersionedDatastreamService" >
```
Comment (bbpennel) [inline services/src/test/resources/retrieve-mods-it-servlet.xml]: Do these need to be here?

File: fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectLoader.java [hunk 5]
```diff
@@ -143,5 +143,4 @@ public RepositoryObject getRepositoryObject(PID pid) {
             }
         }
     }
-
```
Comment (lfarrell) [inline fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectLoader.java]: Did you want to remove this?

Comment (bbpennel) [reply fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectLoader.java]: I had added a new method here and then removed it a commit or two later. I've undone this change now

**Pull request merged**